Neos.ContentRepository.Core/Classes/SharedModel/Node/NodeIdentity.php

```diff
@@ -0,0 +1,93 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Neos\ContentRepository\Core\SharedModel\Node;
+
+use Neos\ContentRepository\Core\DimensionSpace\DimensionSpacePoint;
+use Neos\ContentRepository\Core\Factory\ContentRepositoryId;
+use Neos\ContentRepository\Core\SharedModel\Workspace\WorkspaceName;
+
+/**
+ * The content-repository-id content-stream-id dimension-space-point and the node-aggregate-id
+ * Are used in combination to distinctly identify a single node.
+ *
+ * By using the content graph for the content repository
+ * one can build a subgraph with the right perspective to find this node:
+ *
+ *      $workspace = $contentRepository->getWorkspaceFinder()->findOneByName($identity->workspaceName);
+ *      $subgraph = $contentGraph->getSubgraph(
+ *          $workspace->currentContentStreamId,
+ *          $nodeIdentity->dimensionSpacePoint,
+ *          // resolve also all disabled nodes
+ *          VisibilityConstraints::withoutRestrictions()
+ *      );
+ *      $node = $subgraph->findNodeById($nodeIdentity->nodeAggregateId);
+ *
+ * @api
+ */
+final readonly class NodeIdentity implements \JsonSerializable
+{
+    private function __construct(
+        public ContentRepositoryId $contentRepositoryId,
+        public WorkspaceName $workspaceName,
+        public DimensionSpacePoint $dimensionSpacePoint,
+        public NodeAggregateId $nodeAggregateId,
+    ) {
+    }
+
+    public static function create(
+        ContentRepositoryId $contentRepositoryId,
+        WorkspaceName $workspaceName,
+        DimensionSpacePoint $dimensionSpacePoint,
+        NodeAggregateId $nodeAggregateId,
+    ): self {
+        return new self($contentRepositoryId, $workspaceName, $dimensionSpacePoint, $nodeAggregateId);
+    }
+
+    /**
+     * @param array<string, mixed> $array
+     */
+    public static function fromArray(array $array): self
+    {
+        return new self(
+            ContentRepositoryId::fromString($array['contentRepositoryId']),
+            WorkspaceName::fromString($array['workspaceName']),
+            DimensionSpacePoint::fromArray($array['dimensionSpacePoint']),
+            NodeAggregateId::fromString($array['nodeAggregateId'])
+        );
+    }
+
+    public static function fromJsonString(string $jsonString): self
```

Comment:
Nitpick: JSON is a string by definition so I would suggest to change this to:
```suggestion
    public static function fromJson(string $jsonString): self
```

..or otherwise rename `toJson()` to `toJsonString()` too

Reply:
that naming scheme was not introduce by me but inferred by the codebase ;) See the dsp have the same methods i think

Reply:
Ah OK, dang

Reply:
but i see your point :D Lemme think about it ^^

Reply:
It's not that important, I just stumbled upon it